src/parquet/column/scanner.h

```diff
@@ -167,6 +149,28 @@ class TypedScanner : public Scanner {
   inline void FormatValue(void* val, char* buffer, size_t bufsize, size_t width);
 
   T* values_;
+
+  bool NextLevels(int16_t* def_level, int16_t* rep_level) {
+    bool is_required = def_levels_.size() == 0;
+    if (level_offset_ == levels_buffered_) {
+      int16_t* def_levels_data = is_required ? nullptr : &def_levels_[0];
+      levels_buffered_ = typed_reader_->ReadBatch(batch_size_, def_levels_data,
+          &rep_levels_[0], values_, &values_buffered_);
+
+      // TODO: repetition levels
+
+      value_offset_ = 0;
```

Comment:
it would be best to add a reproduction of this bug in this patch, otherwise another patch will have to be made to resolve PARQUET-502